src/subscribers/abstract_subscriber.jl

```diff
@@ -36,12 +36,13 @@ function got_new!(sub::Subscriber)
     return flags.hasreceived
 end
 
-function decode!(buf::ProtoBuf.ProtoType, bin_data)
-    io = seek(convert(IOStream, bin_data), 0)
-    ProtoBuf.readproto(io, buf)
+function decode!(buf::ProtoBuf.ProtoType, bin_data::IOBuffer)
+    # io = IOBuffer(bin_data)
+    # io = seek(convert(IOStream, bin_data), 0)
+    ProtoBuf.readproto(bin_data, buf)
 end
 
-function decode!(buf::AbstractVector{UInt8}, bin_data)
+function decode!(buf::AbstractVector{UInt8}, bin_data::IOBuffer)
```

Comment:
This isn't a big deal in practice, since `IOBuffer` is a fairly light wrapper around a `Vector{UInt8}`.  You can access the raw buffer with `buffer.data`. I wanted to keep the `decode!` method simple to extend for other custom types. If it took in a `Vector{UInt8}` the protobuf decode would need to allocate a temporary `IOBuffer` (should be  small allocation since it'd just wrap the buffer that's passed in). Either that or it'd have to take the subscriber as an argument, which makes it a bit messier.

Reply:
I have some code in `receive` that automatically (and efficiently) populates the internal `IOBuffer` with the incoming ZMQ data. We could easily do something similar for serial data.